src/azul/openapi/validation.py

```diff
@@ -0,0 +1,109 @@
+import json
+import logging
+from typing import (
+    Mapping,
+)
+
+from chalice import (
+    BadRequestError,
+)
+import jsonschema
+from more_itertools import (
+    one,
+)
+from azul.uuids import (
+    InvalidUUIDError,
+    InvalidUUIDVersionError,
+    validate_uuid,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class SpecValidation:
+    content: Mapping = None
+    in_: str = None
+    name: str = None
+    required: bool = None
+    schema: Mapping = None
+    pattern: str = None
+    validator = None
+
+    def __init__(self, **kwargs):
+        for k, v in kwargs.items():
+            if k == 'in':
+                self.__setattr__("in_", v)
+            else:
+                self.__setattr__(k, v)
+        self.json_validators = dict(jsonschema.Draft7Validator.VALIDATORS)
+
+    def validate(self, request_parameter: Mapping[str, str]):
+        """
+        Validates if a given request parameter meets the spec that is specified for the endpoint.
+        :param request_parameter: A dict with the keys 'name' and 'value'.
+        """
+
+        param_name = request_parameter['name']
+        param_value = request_parameter['value']
+
+        if self.name == 'filters':
+            self.schema = self.content['application/json']['schema']
+            try:
+                filter_json = json.loads(param_value)
+            except Exception:
+                raise BadRequestError(f'The {param_name!r} parameter is not valid JSON')
```

Comment:
This error doesn't have a period at the end of it while the rest all do. Skimming the codebase, it doesn't seem like we have a strict convention on this, but I see a slight preference of no period.

Regardless, I think within a single file, errors should be consistent, so either just add one here, or remove the others.

Reply:
Looking around at other APIs some appear to prefer punctuation and others don't. 
No Punctuation - https://developers.coinbase.com/api/v2#errors
Has Punctuation -  https://developers.google.com/drive/api/v3/handle-errors , https://stripe.com/docs/declines#issuer-declines 

I think we should consistently perform one, just browsing through the [openapi](https://service.dev.singlecell.gi.ucsc.edu/openapi)  endpoint it looks like we have mixed usage. Because these messages are intended to be for humans to read I would be inclined to include the punctuation, (and establish it in the contributing guidelines). There might be some refactoring to do in the API descriptions if we perform this.

@hannes-ucsc do you have input/preference?

Reply:
@hannes-ucsc "no periods if it is a single sentence or fragment. If there are  multiple sentences, then there needs to be a period at the end."

Reply:
@hannes-ucsc  "when making small changes you should follow precedence for what is already there, if you are making large changes you should update the module to the new guidelines."

Reply:
This was already covered. An upcoming PR will add a section title to the guidelines that already cover this. The title will be *Consistency and precedent*